src/bootstrap/compile.rs

```diff
@@ -312,15 +312,18 @@ pub fn std_cargo(builder: &Builder<'_>, target: TargetSelection, stage: u32, car
     // If `compiler-rt` is available ensure that the `c` feature of the
     // `compiler-builtins` crate is enabled and it's configured to learn where
     // `compiler-rt` is located.
+    //
+    // `compiler-rt`'s build script doesn't support cross-compiling, so only enable this for the host.
     let compiler_builtins_root = builder.src.join("src/llvm-project/compiler-rt");
-    let compiler_builtins_c_feature = if compiler_builtins_root.exists() {
-        // Note that `libprofiler_builtins/build.rs` also computes this so if
-        // you're changing something here please also change that.
-        cargo.env("RUST_COMPILER_RT_ROOT", &compiler_builtins_root);
-        " compiler-builtins-c"
-    } else {
-        ""
-    };
+    let compiler_builtins_c_feature =
+        if compiler_builtins_root.exists() && target == builder.build.build {
```

Comment:
This condition is probably overly strict. I'd guess that for e.g. 64-bit -> 32-bit linux the build script probably works fine?

In general, my worry is that this makes it easy for us to have cross-compiled platforms have a fairly different performance (and potentially behavior) profile silently without being able to easily diagnose that...

Reply:
I think to fix the issue in question we can likely have a flag (perhaps off by default, and turned in CI, both in DEPLOY and non-DEPLOY configurations) that gates this rather than detecting based on cross-compilation. We can also give a better error message suggesting to use the flag.

Reply:
Seems reasonable, done. I'm not sure what you meant by a better error message, though - if we were able to predict when this goes wrong, we would be able to fix the bug upstream / avoid using the optimized feature in the first place.

Reply:
I mean that if we are in the condition where we're probably going to fail, we can probably warn about that, or ideally after failure print something like "help: if you want to avoid cross-compiling compiler-builtins C code, enable X flag in config.toml".